File: pure_cnn_py/model/input_layer.py
from pure_cnn_py.model.layer import Layer
from pure_cnn_py.util.mat import Mat
from pure_cnn_py.util.mat_shape import MatShape
from pure_cnn_py.util.constant import INIT_ZEROS, LAYER_INPUT


class InputLayer(Layer):
    def __init__(self, layer_name, image_width, image_height, image_depth, network):
        super(InputLayer, self).__init__(layer_name, 1)
        self.network = network
        self.layer_type = LAYER_INPUT
        self.output_shape = MatShape(image_width, image_height, image_depth)
        self.output = []

    def forward(self, image_data_list):
        for i in range(0, self.network.mini_batch_size):
            self.output.append(Mat(self.output_shape, INIT_ZEROS))
            self.output[i].set_value_by_image(image_data_list[i], self.output_shape.depth)

    def backward(self):
        pass

File: react_mnist/src/model/input_layer.js
import Layer from './layer'
import Mat from '../util/mat'
import Shape from '../util/mat_shape'
import {INIT_ZEROS, LAYER_INPUT} from "../util/constant";


export default class InputLayer extends Layer {
    constructor(layerName, imageWidth, imageHeight, imageDepth) {
        super(layerName, 1);
        this.type = LAYER_INPUT;
        this.output_shape = new Shape(imageWidth, imageHeight, imageDepth);
        this.output = [];
    }

    forward(image_data_list) {
        for (let i = 0; i < this.network.mini_batch_size; i++) {
            this.output[i] = new Mat(this.output_shape, INIT_ZEROS);
            this.output[i].set_value_by_image(image_data_list[i], this.output_shape.depth);
        }
    }

    backward() {
    }
}